R/string_operation.R

```diff
@@ -425,3 +425,11 @@ parse_logical <- function(text, ...){
   }
 }
 
+#'Function to extract text inside the characters like bracket.
+#'@export
+str_extract_inside <- function(column, begin = "(", end = ")") {
+  exp = stringr::str_c("(?<=\\", begin, ").*?(?=\\", end, ")")
```

Comment:
Some reference url about lookahead match and lookbehind match like this as comment would help. https://stackoverflow.com/questions/3926451/how-to-match-but-not-capture-part-of-a-regex

I think the ".*?" in the center of regex can be just ".*".
"?" looks redundant if I'm not missing anything.

Reply:
It seems that the ? after asterisk has effect to handle this case. We'd better keep it. Comment about it would help too.

```
> stringr::str_extract("aaa[xxx][ggg]","(?<=\\[).*(?=\\])")
[1] "xxx][ggg"
> stringr::str_extract("aaa[xxx][ggg]","(?<=\\[).*?(?=\\])")
[1] "xxx"
```

Reply:
Added comment.

Reply:
Please add comment about shortest match too. This is about the ? I was talking in the above comment.
https://stackoverflow.com/questions/1919982/regex-smallest-possible-match-or-nongreedy-match

Reply:
Added comment.